### PR TITLE
Implement the new storage operations for the Multi-Int-Store

### PR DESCRIPTION
**Pull request by jhunzik**
Just wanted to get the draft up early so we're all on the same page. I still need to clean up the tests.

### CODE REVIEW

**Comment by emmberk** [comment]
build now

File: multi-int-store/src/main/java/com/connexta/multiintstore/common/StorageManager.java [hunk 1]
```diff
@@ -6,96 +6,72 @@
...
+    final String key = UUID.randomUUID().toString().replace("-", "");
 
-  private void handleCallback(ProductCallback callback) {
-    LOGGER.info("Product");
-    //  TODO :: Check Markings
-    //  TODO :: Store Product
+    // Store in S3
+    s3.store(mimeType, file, fileSize, fileName, key);
+    // return product location
+    return new URL(retrieveEndpoint + key);
```
Comment (emmberk) [inline multi-int-store/src/main/java/com/connexta/multiintstore/common/StorageManager.java]: ✏️ How should failing to construct the URL be handled?

File: multi-int-store/src/main/java/com/connexta/multiintstore/common/StorageManager.java [hunk 2]
```diff
@@ -6,96 +6,72 @@
...
+  @NotNull private final Dao<IndexedProductMetadata, String> cstDao;
 
-  @Autowired
   public StorageManager(
-      Dao<Product, String> productDao,
-      Dao<IndexedProductMetadata, String> cstDao,
-      DataRetriever retriever) {
-    this.productDao = productDao;
+      @NotEmpty @Value("${endpointUrl.retrieve}") final String retrieveEndpoint,
+      @NotNull final S3StorageAdaptor s3,
```
Comment (emmberk) [inline multi-int-store/src/main/java/com/connexta/multiintstore/common/StorageManager.java]: ✏️ Inject `StorageAdaptor` instead

File: multi-int-store/src/main/java/com/connexta/multiintstore/common/adaptors/S3StorageAdaptor.java [hunk 3]
```diff
@@ -50,9 +48,14 @@ public void store(
...
 
-    LOGGER.info("Storing {} in bucket \"{}\" with key \"{}\"", fileName, s3BucketQuarantine, key);
+    log.info("Storing {} in bucket \"{}\" with key \"{}\"", fileName, s3BucketQuarantine, key);
     try {
       s3Client.putObject(putObjectRequest, requestBody);
+      log.info(
+          "Successfully stored \"{}\" in bucket \"{}\" with key \"{}\"",
+          fileName,
+          s3BucketQuarantine,
+          key);
```
Comment (emmberk) [inline multi-int-store/src/main/java/com/connexta/multiintstore/common/adaptors/S3StorageAdaptor.java]: ✏️ Move this out of the try/catch

File: multi-int-store/src/main/java/com/connexta/multiintstore/common/StorageManager.java (hunk 1 above)
Comment (jhunzik) [reply multi-int-store/src/main/java/com/connexta/multiintstore/common/StorageManager.java]: Probably just the same way we do it in other places: catch a MalformedURLException

File: multi-int-store/src/main/java/com/connexta/multiintstore/common/adaptors/S3StorageAdaptor.java (hunk 3 above)
Comment (jhunzik) [reply multi-int-store/src/main/java/com/connexta/multiintstore/common/adaptors/S3StorageAdaptor.java]: To the end of the method?

Comment (emmberk) [reply multi-int-store/src/main/java/com/connexta/multiintstore/common/adaptors/S3StorageAdaptor.java]: Yes. We don't want to catch runtime exceptions from Logger#info.

Comment (jhunzik) [reply multi-int-store/src/main/java/com/connexta/multiintstore/common/adaptors/S3StorageAdaptor.java]: 🤦‍♂ I should have double checked the exceptions I was catching. Thanks!

File: multi-int-store/src/main/java/com/connexta/multiintstore/common/StorageManager.java (hunk 1 above)
Comment (jhunzik) [reply multi-int-store/src/main/java/com/connexta/multiintstore/common/StorageManager.java]: Actually, we already throw an IOException and the caller logs the message. I think we're ok leaving it as is.

**Comment by ahoffer** [comment]
If the StorageAdaptor is the gateway to blob storage, how about we use blob (or even S3) in the name so that is clear how the class is different than StorageManager? Maybe "BlobStorageManager"?

**Comment by jhunzik** [comment]
build now

**Comment by jhunzik** [comment]
build now (not sure why it failed before)

**Comment by jhunzik** [comment]
build now

**Comment by jhunzik** [comment]
> If the StorageAdaptor is the gateway to blob storage, how about we use blob (or even S3) in the name so that is clear how the class is different than StorageManager? Maybe "BlobStorageManager"?

@ahoffer We kept the adaptor's name generic so that we could change the implementation to support any type of block store. Emily is working on some changes to the StorageManager. Think `ProductStorageManager` and `MetadataStorageManager`

**Comment by emmberk** [comment]
I heroed this locally

**Comment by jhunzik** [comment]
Overriding CI because it's failing to find the owasp dependency check artifact. It passed a full build with owasp locally.

**Pull request merged**